packages/firestore/test/integration/api/get_options.test.ts

```diff
@@ -0,0 +1,612 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { apiDescribe, withTestDoc, withTestCollection } from '../util/helpers';
+
+apiDescribe('GetOptions', persistence => {
+  it('get document while online with default get options', () => {
+    const initialData = { key: 'value' };
+    return withTestDoc(persistence, docRef => {
+      return docRef
+        .set(initialData)
+        .then(() => docRef.get())
+        .then(doc => {
+          expect(doc.exists).to.be.true;
+          expect(doc.metadata.fromCache).to.be.false;
+          expect(doc.metadata.hasPendingWrites).to.be.false;
+          expect(doc.data()).to.deep.equal(initialData);
+        });
+    });
+  });
+
+  it('get collection while online with default get options', () => {
+    const initialDocs = {
+      doc1: { key1: 'value1' },
+      doc2: { key2: 'value2' },
+      doc3: { key3: 'value3' }
+    };
+    return withTestCollection(persistence, initialDocs, colRef => {
+      return colRef.get().then(qrySnap => {
+        expect(qrySnap.metadata.fromCache).to.be.false;
+        expect(qrySnap.metadata.hasPendingWrites).to.be.false;
+        expect(qrySnap.docChanges.length).to.equal(3);
+        let docsData = {};
+        qrySnap.forEach(doc => {
+          docsData[doc.id] = doc.data();
+        });
```

Comment:
We have a toDataArray() helper that does something similar already. Since it stores in an array, it drops the .id, but preserves ordering (which you lose here).

I guess it is convenient to be able to directly `expect(...).to.deep.equal(initialDocs)` though.  So maybe keep this as-is, but consider creating a .toDataMap() helper since you do this in multiple tests?